File: cpp/soma_vetor.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main() {

    int N;
    double soma, media;

    cout << "Quantos numeros voce vai digitar? ";
    cin >> N;

    double vet[N];

    for (int i = 0; i < N; i++) {
        cout << "Digite um numero: ";
        cin >> vet[i];
    }

    soma = 0;
    for (int i = 0; i < N; i++) {
        soma = soma + vet[i];
    }

    media = soma / N;

    cout << fixed << setprecision(1);
    cout << "VALORES = ";

    for (int i = 0; i < N; i++) {
        if (i != 0) {
            cout << " ";
        }
        cout << vet[i];
    }

    cout << fixed << setprecision(2);
    cout << endl << "SOMA = " << soma << endl;
    cout << "MEDIA = " << media << endl;

    return 0;
}

File: java/soma_vetor.java
import java.util.Locale;
import java.util.Scanner;

public class soma_vetor {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N;
		double soma, media;

		System.out.print("Quantos numeros voce vai digitar? ");
		N = sc.nextInt();

		double[] vet = new double[N];

		for (int i = 0; i < N; i++) {
			System.out.print("Digite um numero: ");
			vet[i] = sc.nextDouble();
		}
		
		System.out.println();
		System.out.print("VALORES = ");
		
		for (int i = 0; i < N; i++) {
			System.out.print(String.format("%.1f ", vet[i]));
		}
		System.out.println();

		soma = 0;
		for (int i = 0; i < N; i++) {
			soma = soma + vet[i];
		}
		
		System.out.println("SOMA = " + String.format("%.2f", soma));

		media = soma / N;
		
		System.out.println("MEDIA = " + String.format("%.2f", media));

		sc.close();
	}
}